pkg/config/config.go

```diff
@@ -197,9 +204,23 @@ func (c *MicroshiftConfig) ReadAndValidate(flags *pflag.FlagSet) error {
 		}
 	}
 
+	c.ConfigureClusterURL()
 	return nil
 }
 
+// configures the Cluster URL to have the provided Secure Port
+func (c *MicroshiftConfig) ConfigureClusterURL() {
+	re := regexp.MustCompile(`:(\d+)`)
+	if re.MatchString(c.Cluster.URL) {
```

Comment:
we need to validate both ipv4 and ipv6 addresses, you can reuse `IsValidSocketAddr`

```go
import "k8s.io/apimachinery/pkg/util/validation"

validation.IsValidSocketAddr("1.2.3.4:5678")
```

Reply:
Do you mean validating the addresses after we inject the Secure Port into the URL? 
```golang
// configures the Cluster URL to have the provided Secure Port
func (c *MicroshiftConfig) ConfigureClusterURL() error {
	re := regexp.MustCompile(`:(\d+)`)
	var proposedUrl string
	if re.MatchString(c.Cluster.URL) {
		proposedUrl = re.ReplaceAllString(c.Cluster.URL, fmt.Sprintf(":%d", c.SecurePort))
	} else {
		// if the cluster url doesn't have a port then we must insert it
		re = regexp.MustCompile(`^(\w+)://([\w\.]+)(.*)$`)
		pieces := re.FindStringSubmatch(c.Cluster.URL)
		proposedUrl = pieces[1] + "://" + pieces[2] + ":" + strconv.Itoa(c.SecurePort) + pieces[3]
	}

	// there are errors with our socket address
	errs := validation.IsValidSocketAddr(proposedUrl)
	if 0 < len(errs) {
		return fmt.Errorf("config error: %s", strings.Join(errs, "; "))
	}

	c.Cluster.URL = proposedUrl
	return nil
}
```

Reply:
The usage of `re.MatchString` here is to determine if there is a port within the URL, and if not then to inject the port binding. Although the regex pattern should probably be updated as well so as not to confuse the port number with address bits.

Reply:
I see, that is a good point. Here we should only support `https` (i.e. secure), so `pieces[1]` should be overriden if it is not `https`. Then `validation.IsValidSocketAddr(piece[2] + ":" + strconv.Itoa(c.SecurePort)`  to validate the address format is right. Do you  see a case where `pieces[3]` is not empty?

Reply:
@rootfs Should we be doing the URL validation in this function, or would it make more sense to have a hook that checks the settings post-configuration? I know that this code isn't scalable but my intuition makes me think that we should assume what comes in will be a valid URL, and what we should end up with is a valid URL. 
Could we do those checks within some unit-tests?